AER-1839 Update geoserver to latest version

Gijsvanhorn opened the pull request.

Commit:
update versions and include java version in the pom

Commit:
since version 2.21 the logging filter is configured from the UI, and can be removed from the web.xml. Also logging profiles are now in XML format.

SerhatG commented:
`logging.xml` should refer to the new `xml` config instead of the properties one.
There is also a log file location in the `logging.xml` while the new XML files also contain paths. Not sure if both work or the former is removed/deprecated.

Commit:
use the right extension for the logging profile.
Duplicate log filenames in the logging.xml and the individual profiles seem to be necessary:
* When I remove the fileName from the individual profiles, a log file is never created. Not even when the log file is configured to a different location in the admin interface.
* When I remove the fileName from the global config the field is empty in the admin interface, but the log file is created according to the profile. Then when changing the filename, that new file is created, but the filenames in the individual profiles are not updated.
So for now I will leave the duplication in there...
